### PR TITLE
LINK-1797 | Add signup url to registration serializer

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Add signup_url field to registration serializer so the user knows where the signup form is located

Slack discussion: https://helsinkicity.slack.com/archives/C02L127FC/p1701686920598549

## Closes
[LINK-1797](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1797)

[LINK-1797]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1797?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: registrations/serializers.py [hunk 1]
```diff
@@ -404,6 +407,16 @@ def get_data_source(self, obj):
...
 
+    def get_signup_url(self, obj):
+        def _get_signup_url_for_lang(lang):
+            return f"{settings.LINKED_REGISTRATIONS_UI_URL}/{lang}/registration/{obj.id}/signup-group/create"
+
+        return {
+            "en": _get_signup_url_for_lang("en"),
+            "fi": _get_signup_url_for_lang("fi"),
+            "sv": _get_signup_url_for_lang("sv"),
+        }
```
Comment (harriris-vincit) [inline registrations/serializers.py:418]:
```suggestion
        return {
            lang: f"{settings.LINKED_REGISTRATIONS_UI_URL}/{lang}/registration/{obj.id}/signup-group/create"
            for lang in ["en", "fi", "sv"]
        }
```

**Comment by harriris-vincit** [comment]
A test for the expected content of the URL would be a good addition. Otherwise looks good

File: registrations/serializers.py (hunk 1 above)
Comment (jorilindell) [reply registrations/serializers.py]: Changed as suggested and added a test for the content

File: registrations/tests/test_registration_get.py [hunk 2]
```diff
@@ -221,6 +223,16 @@ def test_anonymous_user_can_see_registration(api_client, registration):
     assert_registration_fields_exist(response.data, is_admin_user=False)
 
 
+@pytest.mark.django_db
+def test_get_registration_contains_correct_signup_url(user_api_client, registration):
+    response = get_detail_and_assert_registration(user_api_client, registration.id)
+    for lang in ["en", "fi", "sv"]:
```
Comment (harriris-vincit) [inline registrations/tests/test_registration_get.py:229]: Testing for complete equality (`response.json() == <urls_dict>`) would ensure that there are no extra keys/data

Comment (jorilindell) [reply registrations/tests/test_registration_get.py]: Modified the test to assert that `signup_url` doesn't contain any extra fields

**Review by harriris-vincit — APPROVED** [review]
LGTM!

**Review by tuomas777 — APPROVED** [review]

**Pull request merged**